lib/teacher_services_tech_docs/github/ruby_dependencies.rb

```diff
@@ -1,32 +1,61 @@
 module SchoolsDigitalTechDocs
   module GitHub
     class RubyDependencies
-      def initialize(lockfile_contents)
-        @parsed_lockfile = Bundler::LockfileParser.new(lockfile_contents)
+      def initialize(service_name, lockfile:, tool_versions_file: nil, ruby_version_file: nil)
+        @service_name = service_name
+        @lockfile = lockfile
+        @tool_versions_file = tool_versions_file
+        @ruby_version_file = ruby_version_file
       end
 
       def rails_version
-        @parsed_lockfile.specs.find { |s| s.name == "rails" }.version.to_s
+        get_dependency_version("rails")
       end
 
       def dfe_analytics_version
-        @parsed_lockfile.specs.find { |s| s.name == "dfe-analytics" }&.version&.to_s
+        get_dependency_version("dfe-analytics")
       end
 
       def dfe_autocomplete_version
-        @parsed_lockfile.specs.find { |s| s.name == "dfe-autocomplete" }&.version&.to_s
+        get_dependency_version("dfe-autocomplete")
       end
 
       def dfe_reference_data_version
-        @parsed_lockfile.specs.find { |s| s.name == "dfe-reference-data" }&.version&.to_s
+        get_dependency_version("dfe-reference-data")
+      end
+
+      def has_tool_versions
```

Comment:
Maybe add a `?` suffix for this predicate method?
```suggestion
      def has_tool_versions?
```